fuzzywuzzy/process.py

```diff
@@ -165,11 +165,17 @@ def extract(query, choices, processor=default_processor, scorer=default_scorer,
         [('train', 22, 'bard'), ('man', 0, 'dog')]
     """
     sl = extractWithoutOrder(query, choices, processor, scorer)
-    return heapq.nlargest(limit, sl, key=lambda i: i[1]) if limit is not None else \
-        sorted(sl, key=lambda i: i[1], reverse=True)
+    if char_sort is False:
+        return heapq.nlargest(limit, sl, key=lambda i: i[1]) if limit is not None else \
+            sorted(sl, key=lambda i: i[1], reverse=True)
+    else:
+        best_list = sorted(sl, key=lambda i: i[1], reverse=True)
+        return sortByCommonLetter(sl, query)[0: min(limit, len(sl))] if limit is not None else \
```

Comment:
from a performance standpoint this is a bad idea. When the user is only interested in 5 elements (the default), but has e.g. 1 million choices, this will sort 1 million choices (with a slow algorithm, since it counts all the letters) and then only takes the best 5 elements.